api/handler/delete.go

```diff
@@ -128,34 +138,48 @@ func (h *handler) DeleteMultipleObjectsHandler(w http.ResponseWriter, r *http.Re
 		return nil
 	})
 
-	if errs := h.obj.DeleteObjects(r.Context(), reqInfo.BucketName, toRemove); errs != nil && !requested.Quiet {
-		additional := []zap.Field{
-			zap.Array("objects", marshaler),
-			zap.Errors("errors", errs),
+	deletedObjects, err := h.obj.DeleteObjects(r.Context(), reqInfo.BucketName, toRemove)
+	if !requested.Quiet {
+		if err != nil {
+			h.logAndSendError(w, "couldn't delete objects", reqInfo, err)
+			return
 		}
-		h.logAndSendError(w, "could not delete objects", reqInfo, nil, additional...)
 
-		for _, e := range errs {
-			if err, ok := e.(*errors.ObjectError); ok {
+		var errs []error
+		for _, obj := range deletedObjects {
+			if obj.Error == nil {
+				deletedObj := DeletedObject{
+					ObjectIdentifier: ObjectIdentifier{
+						ObjectName: obj.Name,
+						VersionID:  obj.VersionID,
+					},
+					DeleteMarkerVersionID: obj.DeleteMarkVersion,
+				}
+				if deletedObj.DeleteMarkerVersionID != "" {
+					deletedObj.DeleteMarker = true
+				}
+				response.DeletedObjects = append(response.DeletedObjects, deletedObj)
+			} else if !requested.Quiet {
```

Comment:
I guess it is always true because of line 142.

Reply:
Oh, that must be the end bracket of the condition (line 142) is misplaced